dnf-behave-tests/dnf/steps/server.py

```diff
@@ -107,3 +107,10 @@ def step_http_log_contains(context):
     for e in expected:
         if e not in found:
             raise AssertionError('HTTP log does not contain "%s": %s' % (e, "\n" + "\n".join(found) + "\n"))
+
+
+@behave.step("HTTP server GET {url} response headers are")
+def step_http_server_get_response_headers(context, url):
+    headers = [tuple(context.text.split('=', 1))]
+    context.scenario.httpd.conf["status"] = 404
```

Comment:
This line should not be here, it is again specific for your scenario, where you need to return 404 for your particular case. Remove this and instead add this step directly to your scenario:
```
And the server starts responding with HTTP status code 404
```
That should do it for you :slightly_smiling_face: